src/azul/es.py

```diff
@@ -1,24 +1,25 @@
 from elasticsearch import Elasticsearch, RequestsHttpConnection
-from aws_requests_auth import boto_utils
-from aws_requests_auth.aws_auth import AWSRequestsAuth
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth
+from functools import lru_cache
 
 from azul import config
 
 
-def es_client():
-    host = config.es_endpoint[0]
-    port = config.es_endpoint[1]
-    ssl_required = True if port == 443 else False
-    if ssl_required:
-        aws_auth = AWSRequestsAuth(aws_host=host,
-                                   aws_region='us-east-1',
-                                   aws_service='es',
-                                   **boto_utils.get_credentials())
-        return Elasticsearch(hosts=[{'host': host, 'port': port}],
-                             http_auth=aws_auth,
-                             use_ssl=True,
-                             verify_certs=True,
-                             connection_class=RequestsHttpConnection)
-    else:
-        return Elasticsearch(hosts=[{'host': host, 'port': port}],
-                             connection_class=RequestsHttpConnection)
+class ElasticsearchClient:
+
+    @staticmethod
+    def get():
+        host, port = config.es_endpoint
+        ssl_required = True if port == 443 and host.endswith(".amazonaws.com") else False
+        return ElasticsearchClient._create_client(host, port, ssl_required, 60)
+
+    @staticmethod
+    @lru_cache(maxsize=32)
+    def _create_client(host, port, ssl_required, timeout):
+        common_params = dict(hosts=[dict(host=host, port=port)], timeout=timeout)
+        if ssl_required:
+            aws_auth = BotoAWSRequestsAuth(aws_host=host, aws_region='us-east-1', aws_service='es')
```

Comment:
The hard-coded region is actually the cause for https://github.com/DataBiosphere/azul/issues/194. Can you be in touch with @achave11 because he has a fix for this but no PR yet.